<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [Mono-list] mono clr via pythonnet: invalid pointer
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mono-list%40lists.dot.net?Subject=Re%3A%20%5BMono-list%5D%20mono%20clr%20via%20pythonnet%3A%20invalid%20pointer&In-Reply-To=%3CCACaWZ9SCprFN%2BCiz74%3Dx%2BYA%3D3-U9Pbpg6-fDdirydz9A2OUX6g%40mail.gmail.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=utf-8">
   <LINK REL="Previous"  HREF="052338.html">
   <LINK REL="Next"  HREF="052344.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mono-list] mono clr via pythonnet: invalid pointer</H1>
    <B>William Ivanski</B> 
    <A HREF="mailto:mono-list%40lists.dot.net?Subject=Re%3A%20%5BMono-list%5D%20mono%20clr%20via%20pythonnet%3A%20invalid%20pointer&In-Reply-To=%3CCACaWZ9SCprFN%2BCiz74%3Dx%2BYA%3D3-U9Pbpg6-fDdirydz9A2OUX6g%40mail.gmail.com%3E"
       TITLE="[Mono-list] mono clr via pythonnet: invalid pointer">william.ivanski at gmail.com
       </A><BR>
    <I>Wed Jan 18 10:56:03 UTC 2017</I>
    <P><UL>
        <LI>Previous message (by thread): <A HREF="052338.html">[Mono-list] mono clr via pythonnet: invalid pointer
</A></li>
        <LI>Next message (by thread): <A HREF="052344.html">[Mono-list] mono clr via pythonnet: invalid pointer
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#52339">[ date ]</a>
              <a href="thread.html#52339">[ thread ]</a>
              <a href="subject.html#52339">[ subject ]</a>
              <a href="author.html#52339">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Hi, Andres, Daniel and others,

It works, I'm using Kubuntu 16.10 and mono 4.6.2. To run:

mono nPython.exe test.py

I compiled Python for .NET myself from GitHub repo. You can always toss
away the EXE and the DLL, and compile them yourself.

Em qua, 18 de jan de 2017 às 03:16, Andres &lt;<A HREF="http://lists.dot.net/mailman/listinfo/mono-list">knocte at gmail.com</A>&gt; escreveu:

&gt;<i> Daniel, and others, I advice not to consume random binaries from random
</I>&gt;<i> people on the internet.
</I>&gt;<i>
</I>&gt;<i> On Wednesday, January 18, 2017 06:09 AM, William Ivanski wrote:
</I>&gt;<i> &gt; Please see solution
</I>&gt;<i> &gt; here:
</I>&gt;<i> <A HREF="https://drive.google.com/file/d/0B6qR86JfcfthdDNwS2g4V2RFRmc/view?usp=sharing">https://drive.google.com/file/d/0B6qR86JfcfthdDNwS2g4V2RFRmc/view?usp=sharing</A>
</I>&gt;<i> &gt;
</I>&gt;<i> &gt; It works, I'm using Kubuntu 16.10 and mono 4.6.2. I compiled Python for
</I>&gt;<i> &gt; .NET myself from GitHub repo.
</I>&gt;<i> &gt; To run:
</I>&gt;<i> &gt;
</I>&gt;<i> &gt; mono nPython.exe test.py
</I>&gt;<i> &gt;
</I>&gt;<i> &gt; Em ter, 17 de jan de 2017 às 18:57, Daniel Krause &lt;<A HREF="http://lists.dot.net/mailman/listinfo/mono-list">madakr at web.de</A>
</I>&gt;<i> &gt; &lt;mailto:<A HREF="http://lists.dot.net/mailman/listinfo/mono-list">madakr at web.de</A>&gt;&gt; escreveu:
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Hello,
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     I am trying to to use mono from python via pythonnet.
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     My system:
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     *Ubuntu*
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;       * Release 16.04.1 LTS (Xenial Xerus) 32-bit
</I>&gt;<i> &gt;       * Kernel Linux 4.1.19-v7+ armv7l
</I>&gt;<i> &gt;       * MATE 1.12.1
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     *Hardware*
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;       * Memory: 925,8 MiB
</I>&gt;<i> &gt;       * Processor: ARMv7 Processor rev 4 (v7l) × 4
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     *Mono: *mono-complete (4.2.4.4-0wheezy1)
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     **More information on the system and installed dependencies for
</I>&gt;<i> &gt;     pythonnet can be found in this thread:
</I>&gt;<i> &gt;     <A HREF="https://github.com/pythonnet/pythonnet/issues/327">https://github.com/pythonnet/pythonnet/issues/327</A>
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     The following small test script fails at the first step (importing
</I>&gt;<i> &gt;     the clr):
</I>&gt;<i> &gt;     #!/usr/bin/env python3
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     print('import clr')
</I>&gt;<i> &gt;     import clr
</I>&gt;<i> &gt;     print('clr imported')
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     if __name__ == '__main__':
</I>&gt;<i> &gt;         print('main')
</I>&gt;<i> &gt;         exit(0)
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     It gives the following traceback:
</I>&gt;<i> &gt;     import clr
</I>&gt;<i> &gt;     *** Error in `python3.5': free(): invalid pointer: 0x75b92050 ***
</I>&gt;<i> &gt;     Stacktrace:
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;       at &lt;unknown&gt; &lt;0xffffffff&gt;
</I>&gt;<i> &gt;       at (wrapper managed-to-native)
</I>&gt;<i> &gt;     Python.Runtime.Runtime.Py_Initialize () &lt;0xffffffff&gt;
</I>&gt;<i> &gt;       at Python.Runtime.Runtime.Initialize () &lt;0x0002f&gt;
</I>&gt;<i> &gt;       at Python.Runtime.PythonEngine.Initialize () &lt;0x00093&gt;
</I>&gt;<i> &gt;       at Python.Runtime.PythonEngine.InitExt () &lt;0x0002f&gt;
</I>&gt;<i> &gt;       at (wrapper runtime-invoke) &lt;Module&gt;.runtime_invoke_intptr
</I>&gt;<i> &gt;     (object,intptr,intptr,intptr) &lt;0xffffffff&gt;
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Native stacktrace:
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Debug info from gdb:
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     [New LWP 1901]
</I>&gt;<i> &gt;     [Thread debugging using libthread_db enabled]
</I>&gt;<i> &gt;     Using host libthread_db library
</I>&gt;<i> &gt;     &quot;/lib/arm-linux-gnueabihf/libthread_db.so.1&quot;.
</I>&gt;<i> &gt;     0x76f23454 in __libc_do_syscall () from
</I>&gt;<i> &gt;     /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;       Id   Target Id         Frame
</I>&gt;<i> &gt;     * 1    Thread 0x76f8a300 (LWP 1900) &quot;python3.5&quot; 0x76f23454 in
</I>&gt;<i> &gt;     __libc_do_syscall () from /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;       2    Thread 0x76250470 (LWP 1901) &quot;Finalizer&quot; 0x76f23454 in
</I>&gt;<i> &gt;     __libc_do_syscall () from /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Thread 2 (Thread 0x76250470 (LWP 1901)):
</I>&gt;<i> &gt;     #0  0x76f23454 in __libc_do_syscall () from
</I>&gt;<i> &gt;     /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;     #1  0x76f20834 in do_futex_wait.constprop () from
</I>&gt;<i> &gt;     /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;     #2  0x76f20916 in __new_sem_wait_slow.constprop.0 () from
</I>&gt;<i> &gt;     /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;     #3  0x767dae84 in mono_sem_wait () from
</I>&gt;<i> /usr/lib/libmonoboehm-2.0.so.1
</I>&gt;<i> &gt;     #4  0x767a89a0 in ?? () from /usr/lib/libmonoboehm-2.0.so.1
</I>&gt;<i> &gt;     Backtrace stopped: previous frame identical to this frame (corrupt
</I>&gt;<i> &gt;     stack?)
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Thread 1 (Thread 0x76f8a300 (LWP 1900)):
</I>&gt;<i> &gt;     #0  0x76f23454 in __libc_do_syscall () from
</I>&gt;<i> &gt;     /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;     #1  0x76f2232a in waitpid () from
</I>&gt;<i> &gt;     /lib/arm-linux-gnueabihf/libpthread.so.0
</I>&gt;<i> &gt;     #2  0x767122bc in ?? () from /usr/lib/libmonoboehm-2.0.so.1
</I>&gt;<i> &gt;     Backtrace stopped: previous frame identical to this frame (corrupt
</I>&gt;<i> &gt;     stack?)
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     =================================================================
</I>&gt;<i> &gt;     Got a SIGABRT while executing native code. This usually indicates
</I>&gt;<i> &gt;     a fatal error in the mono runtime or one of the native libraries
</I>&gt;<i> &gt;     used by your application.
</I>&gt;<i> &gt;     =================================================================
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Aborted (core dumped)
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     My debugging skills are very poor, anyway, I tried to use valgrind
</I>&gt;<i> &gt;     to get perhaps a better idea. The output:
</I>&gt;<i> &gt;     ==2813== Memcheck, a memory error detector
</I>&gt;<i> &gt;     ==2813== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et
</I>&gt;<i> al.
</I>&gt;<i> &gt;     ==2813== Using Valgrind-3.11.0 and LibVEX; rerun with -h for
</I>&gt;<i> &gt;     copyright info
</I>&gt;<i> &gt;     ==2813== Command: python3.5 pythonnet_test.py
</I>&gt;<i> &gt;     ==2813==
</I>&gt;<i> &gt;     disInstr(arm): unhandled instruction: 0xF1010200
</I>&gt;<i> &gt;                      cond=15(0xF) 27:20=16(0x10) 4:4=0 3:0=0(0x0)
</I>&gt;<i> &gt;     ==2813== valgrind: Unrecognised instruction at address 0x48596f4.
</I>&gt;<i> &gt;     ==2813==    at 0x48596F4: ??? (in
</I>&gt;<i> &gt;     /usr/lib/arm-linux-gnueabihf/libarmmem.so)
</I>&gt;<i> &gt;     ==2813== Your program just tried to execute an instruction that
</I>&gt;<i> Valgrind
</I>&gt;<i> &gt;     ==2813== did not recognise.  There are two possible reasons for this.
</I>&gt;<i> &gt;     ==2813== 1. Your program has a bug and erroneously jumped to a
</I>&gt;<i> non-code
</I>&gt;<i> &gt;     ==2813==    location.  If you are running Memcheck and you just saw a
</I>&gt;<i> &gt;     ==2813==    warning about a bad jump, it's probably your program's
</I>&gt;<i> &gt;     fault.
</I>&gt;<i> &gt;     ==2813== 2. The instruction is legitimate but Valgrind doesn't
</I>&gt;<i> &gt;     handle it,
</I>&gt;<i> &gt;     ==2813==    i.e. it's Valgrind's fault.  If you think this is the
</I>&gt;<i> &gt;     case or
</I>&gt;<i> &gt;     ==2813==    you are not sure, please let us know and we'll try to
</I>&gt;<i> &gt;     fix it.
</I>&gt;<i> &gt;     ==2813== Either way, Valgrind will now raise a SIGILL signal which
</I>&gt;<i> will
</I>&gt;<i> &gt;     ==2813== probably kill your program.
</I>&gt;<i> &gt;     ==2813==
</I>&gt;<i> &gt;     ==2813== Process terminating with default action of signal 4 (SIGILL)
</I>&gt;<i> &gt;     ==2813==  Illegal opcode at address 0x48596F4
</I>&gt;<i> &gt;     ==2813==    at 0x48596F4: ??? (in
</I>&gt;<i> &gt;     /usr/lib/arm-linux-gnueabihf/libarmmem.so)
</I>&gt;<i> &gt;     ==2813==
</I>&gt;<i> &gt;     ==2813== HEAP SUMMARY:
</I>&gt;<i> &gt;     ==2813==     in use at exit: 1,088 bytes in 10 blocks
</I>&gt;<i> &gt;     ==2813==   total heap usage: 62 allocs, 52 frees, 5,692 bytes
</I>&gt;<i> allocated
</I>&gt;<i> &gt;     ==2813==
</I>&gt;<i> &gt;     ==2813== LEAK SUMMARY:
</I>&gt;<i> &gt;     ==2813==    definitely lost: 0 bytes in 0 blocks
</I>&gt;<i> &gt;     ==2813==    indirectly lost: 0 bytes in 0 blocks
</I>&gt;<i> &gt;     ==2813==      possibly lost: 0 bytes in 0 blocks
</I>&gt;<i> &gt;     ==2813==    still reachable: 1,088 bytes in 10 blocks
</I>&gt;<i> &gt;     ==2813==         suppressed: 0 bytes in 0 blocks
</I>&gt;<i> &gt;     ==2813== Rerun with --leak-check=full to see details of leaked memory
</I>&gt;<i> &gt;     ==2813==
</I>&gt;<i> &gt;     ==2813== For counts of detected and suppressed errors, rerun with: -v
</I>&gt;<i> &gt;     ==2813== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from
</I>&gt;<i> 0)
</I>&gt;<i> &gt;     Illegal instruction (core dumped)
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Both the traceback from python and the output of valgrind are beyond
</I>&gt;<i> &gt;     my comprehension.
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Has anyone have an idea, what is wrong, and how to fix it?
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;     Thanks
</I>&gt;<i> &gt;     Daniel
</I>&gt;<i> &gt;     _______________________________________________
</I>&gt;<i> &gt;     Mono-list maillist  -  <A HREF="http://lists.dot.net/mailman/listinfo/mono-list">Mono-list at lists.dot.net</A>
</I>&gt;<i> &gt;     &lt;mailto:<A HREF="http://lists.dot.net/mailman/listinfo/mono-list">Mono-list at lists.dot.net</A>&gt;
</I>&gt;<i> &gt;     <A HREF="http://lists.dot.net/mailman/listinfo/mono-list">http://lists.dot.net/mailman/listinfo/mono-list</A>
</I>&gt;<i> &gt;
</I>&gt;<i> &gt; --
</I>&gt;<i> &gt;
</I>&gt;<i> &gt; William Ivanski - Microsoft MVP
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;
</I>&gt;<i> &gt;
</I>&gt;<i> &gt; _______________________________________________
</I>&gt;<i> &gt; Mono-list maillist  -  <A HREF="http://lists.dot.net/mailman/listinfo/mono-list">Mono-list at lists.dot.net</A>
</I>&gt;<i> &gt; <A HREF="http://lists.dot.net/mailman/listinfo/mono-list">http://lists.dot.net/mailman/listinfo/mono-list</A>
</I>&gt;<i> &gt;
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i> _______________________________________________
</I>&gt;<i> Mono-list maillist  -  <A HREF="http://lists.dot.net/mailman/listinfo/mono-list">Mono-list at lists.dot.net</A>
</I>&gt;<i> <A HREF="http://lists.dot.net/mailman/listinfo/mono-list">http://lists.dot.net/mailman/listinfo/mono-list</A>
</I>&gt;<i>
</I>-- 

William Ivanski - Microsoft MVP
-------------- next part --------------
An HTML attachment was scrubbed...
URL: &lt;<A HREF="http://lists.dot.net/pipermail/mono-list/attachments/20170118/63fd7e42/attachment-0001.html">http://lists.dot.net/pipermail/mono-list/attachments/20170118/63fd7e42/attachment-0001.html</A>&gt;
</PRE>



<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message (by thread): <A HREF="052338.html">[Mono-list] mono clr via pythonnet: invalid pointer
</A></li>
	<LI>Next message (by thread): <A HREF="052344.html">[Mono-list] mono clr via pythonnet: invalid pointer
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#52339">[ date ]</a>
              <a href="thread.html#52339">[ thread ]</a>
              <a href="subject.html#52339">[ subject ]</a>
              <a href="author.html#52339">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.dot.net/mailman/listinfo/mono-list">More information about the Mono-list
mailing list</a><br>
</body></html>
